src/meta/processors/job/JobManager.cpp

```diff
@@ -148,8 +140,11 @@ void JobManager::scheduleThread() {
 
 // @return: true if all task dispatched, else false
 bool JobManager::runJobInternal(const JobDescription& jobDesc, JbOp op) {
-  std::unique_ptr<MetaJobExecutor> jobExec =
-      MetaJobExecutorFactory::createMetaJobExecutor(jobDesc, kvStore_, adminClient_);
+  std::lock_guard<std::mutex> lk(muJobFinished_);
+  std::unique_ptr<JobExecutor> je =
+      JobExecutorFactory::createJobExecutor(jobDesc, kvStore_, adminClient_);
+  JobExecutor* jobExec = je.get();
```

Comment:
Why not use shared_ptr? If the je has been destruct other place, it will casuse some problems.